contents/code/main.js

```diff
@@ -1,143 +1,136 @@
+const MIN_DESKTOPS = 2;
+
 function add_desktop()
 {
+	print("add_desktop()");
 	workspace.desktops += 1;
 }
 
 // shifts a window to the left if it's more to the right than number
-function shift_righter_than(number)
+function shift_righter_than(client, number)
 {
-	return function(client)
+	if (client.desktop > number)
 	{
-		if (client.desktop > number) {
-			client.desktop -= 1;
-		}
+		print(`Shifting ${client.caption} to desktop ${client.desktop - 1}`);
+		client.desktop -= 1;
 	}
 }
 
-// deletes last desktop without fanfare
-function delete_last()
+/**
+ * Delete a desktop by number
+ * Returns true if desktop was deleted, false if wasn't
+ * @returns true if removed
+ */
+function remove_desktop_with(number)
 {
-	// don't do anything if last two remain
-	if (workspace.desktops <= 2)  return;
-	workspace.desktops -= 1;
-}
+	print(`remove_desktop_with(${number})`);
 
-// simulates deletion of desktop in the middle
-// Returns true if desktop was deleted, false if wasn't
-function delete_desktop(number)
-{
-	print("delete desktop " + number);
-	// don't do anything for last desktop
-	if (workspace.desktops == 1)  return false;
-	if (workspace.desktops == number)  return false;
-
-	if (workspace.desktops == 2) {
-		// don't delete, only shift left
-		workspace.clientList().forEach(shift_righter_than(number));
-		return false;
-	}
+	// don't do anything if below minimum desktops
+	if (workspace.desktops <= MIN_DESKTOPS) return false;
 
-	if (number >= workspace.desktops - 1) {
-		// delete without shifting
-		delete_last();
-		return true;
-	}
+	// do not remove empty desktop at the end
+	if (workspace.desktops == number) return false;
+
+	// Shift all clients right from desktop $number to the left
+	// instead of deleting the desktop directly
+	// Once shifted remove desktop with the highest number
+	// This is less efficient than deleting directly,
+	// BUT the workspace names do not get messed up over time
```

Comment:
It's not that it's written like this for some specific reason, it's just that there isn't (wasn't?) any way to delete a desktop in the middle when I wrote this

Reply:
I wonder, do users of this script even care about desktop names? I personally don't. This whole function can now be replaced with the new removeDesktop method

Reply:
It is possible, but it makes anyway sense to do it like that, otherwise you have weird mixed up desktop names like : desktop 1, desktop 5, desktop 3, etc....